Rubberduck.VBEEditor/ComManagement/StrongComSafe.cs

```diff
@@ -1,32 +1,43 @@
-﻿using System;
-using System.Collections.Concurrent;
+﻿using System.Collections.Concurrent;
+using System.Collections.Generic;
 using Rubberduck.VBEditor.SafeComWrappers.Abstract;
 
 namespace Rubberduck.VBEditor.ComManagement
 {
-    public class StrongComSafe: IComSafe
+    public class StrongComSafe: ComSafeBase
     {
         //We override the equality comparison and hash code because subclasses of SafeComWrapper<T> override the corresponding methods.
         //We need to distinguish between the individual wrapper instances no matter whether they are semantically equal.
-        private readonly ConcurrentDictionary<ISafeComWrapper, byte> _comWrapperCache = new ConcurrentDictionary<ISafeComWrapper, byte>(new ReferenceEqualityComparer());
+        private readonly ConcurrentDictionary<int, ISafeComWrapper> _comWrapperCache = new ConcurrentDictionary<int, ISafeComWrapper>();
```

Comment:
The change was made primarily to simplify the implementation of `GetWrappers` and encourage consistency in how we hash between the 2 versions of safe. If this is unwanted, please let me know.

Reply:
I do not like this change because it removes the safety net for hash collisions. (When encountering the same hash, an equality check gets performed.) Having the wrapper as a key is the clean way to simulate a `ConcurrentHashSet`. Unfortunately, this is not possible in the weak COM safe because it requires a hard reference to the wrapper.

Reply:
Ok, will revert that and modify the `GetWrappers` method to provide the hashcode. Does that imply that the weak COM safe lacks that safety net?

Reply:
Yes, the weak COM safe does not have the safety net because it is impossible to have it without holding a strong reference to the SCW.